src/tools/clippy/clippy_lints/src/float_literal.rs

```diff
@@ -139,10 +145,11 @@ impl<'tcx> LateLintPass<'tcx> for FloatLiteral {
 #[must_use]
 fn max_digits(fty: FloatTy) -> u32 {
     match fty {
-        FloatTy::F16 => unimplemented!("f16_f128"),
+        // FIXME(f16_f128): return a real value once core is available
+        FloatTy::F16 => 0,
```

Comment:
Just in case that the PR that this relies on gets to master before Clippy, could you change this? And change the FIXME to refer to f16::DIGITS, instead of the magic number

```suggestion
        FloatTy::F16 => 3,
```